aten/src/ATen/native/TensorIteratorReduce.cpp

```diff
@@ -117,13 +117,14 @@ static void parallel_dim_reduction(TensorIterator& iter, const loop2d_t& loop) {
 }
 
 void TensorIterator::foreach_reduced_elt(const loop_subiter_t &loop, bool parallelize) {
-  AT_ASSERT(ntensors() == 2 && num_outputs_ == 1);
+  AT_ASSERT(ntensors() - num_outputs_ == 1); //Number of inputs = 1
```

Comment:
We should add an `ninputs` function to TensorIterator that does this computation

Reply:
Added ninputs function to TensorIterator